packages/imperative/src/events/src/EventUtils.ts

```diff
@@ -0,0 +1,199 @@
+     * @return {boolean} True if it is a user event, otherwise false.
+     */
+    public static isUserEvent(eventName: string): boolean {
+        return Object.values<string>(ZoweUserEvents).includes(eventName);
+    }
+
+    /**
+     * Determines if the specified event name is associated with a shared event.
+     *
+     * @param {string} eventName - The name of the event.
+     * @return {boolean} True if it is a shared event, otherwise false.
+     */
+    public static isSharedEvent(eventName: string): boolean {
+        return Object.values<string>(ZoweSharedEvents).includes(eventName);
+    }
+
+    /**
+     * Determines the directory path for storing event files based on the event type and application name.
+     *
+     * @param {EventTypes} eventType - The type of event.
+     * @param {string} appName - The name of the application.
+     * @return {string} The directory path.
+     */
+    public static getEventDir(eventType: EventTypes, appName: string): string {
+        this.validateAppName(appName);
+        return eventType === EventTypes.SharedEvents || eventType === EventTypes.UserEvents ?
+            join(".events", appName) : ".events";
+    }
+
+    /**
+     * Ensures that the specified directory for storing event files exists, creating it if necessary.
+     *
+     * @param {string} directoryPath - The path to the directory.
+     */
+    public static ensureEventsDirExists(directoryPath: string) {
+        try {
+            if (!fs.existsSync(directoryPath)) {
+                fs.mkdirSync(directoryPath);
+            }
+        } catch (err) {
+            throw new ImperativeError({ msg: `Unable to create '.events' directory. Path: ${directoryPath}`, causeErrors: err });
+        }
+    }
+
+    /**
+     * Ensures that the specified file path for storing event data exists, creating it if necessary.
+     *
+     * @param {string} filePath - The path to the file.
+     */
+    public static ensureFileExists(filePath: string) {
+        try {
+            if (!fs.existsSync(filePath)) {
+                fs.closeSync(fs.openSync(filePath, 'w'));
+            }
+        } catch (err) {
+            throw new ImperativeError({ msg: `Unable to create event file. Path: ${filePath}`, causeErrors: err });
+        }
+    }
+
+    /**
+     * Creates and registers a new event subscription for a specific event processor.
+     *
+     * @param {EventProcessor} eeInst - The event processor instance.
+     * @param {string} eventName - The name of the event.
+     * @param {EventTypes} eventType - The type of event.
+     * @return {IEventDisposable} An interface for managing the subscription.
+     */
+    public static createSubscription(eeInst: EventProcessor, eventName: string, eventType: EventTypes): IEventDisposable {
+        const dir = this.getEventDir(eventType, eeInst.appName);
+        this.ensureEventsDirExists(join(ConfigUtils.getZoweDir(), '.events'));
```

Comment:
Is this line redundant, since we make sure that the events dir exists in the `ensureEventsDirExists` call on the next line?

Reply:
We are not doing a `mkdir -p`.
I guess we could just replace the way `ensureEventDirExists` works and allow for directory creation to be recursive 😋